modules/v2/variables.tf

```diff
@@ -0,0 +1,313 @@
+  default     = {}
+}
+
+variable "template_annotations" {
+  type        = map(string)
+  description = "Unstructured key value map that may be set by external tools to store and arbitrary metadata. They are not queryable and should be preserved when modifying objects. Refer https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/cloud_run_v2_service#annotations"
+  default     = {}
+}
+
+variable "timeout" {
+  type        = string
+  description = "Max allowed time for an instance to respond to a request. A duration in seconds with up to nine fractional digits, ending with 's'"
+  default     = null
+}
+
+variable "service_account" {
+  type        = string
+  description = "Email address of the IAM service account associated with the revision of the service"
+  default     = null
+}
+
+variable "encryption_key" {
+  description = "A reference to a customer managed encryption key (CMEK) to use to encrypt this container image."
+  type        = string
+  default     = null
+}
+
+variable "max_instance_request_concurrency" {
+  type        = string
+  description = "Sets the maximum number of requests that each serving instance can receive"
+  default     = null
+}
+
+variable "session_affinity" {
+  type        = string
+  description = "Enables session affinity. For more information, go to https://cloud.google.com/run/docs/configuring/session-affinity"
+  default     = null
+}
+
+variable "execution_environment" {
+  type        = string
+  description = "The sandbox environment to host this Revision."
+  default     = "EXECUTION_ENVIRONMENT_GEN2"
+
+  validation {
+    condition     = contains(["EXECUTION_ENVIRONMENT_GEN1", "EXECUTION_ENVIRONMENT_GEN2"], var.execution_environment)
+    error_message = "Allowed values for ingress are \"EXECUTION_ENVIRONMENT_GEN1\", \"EXECUTION_ENVIRONMENT_GEN2\"."
+  }
+}
+
+variable "volumes" {
+  type = list(object({
+    name = string
+    secret = object({
+      secret       = string
+      default_mode = optional(string)
+      items = optional(object({
+        path    = string
+        version = optional(string)
+        mode    = optional(string)
+      }))
+    })
+    cloud_sql_instance = optional(object({
+      instances = optional(string)
+    }))
+    empty_dir = optional(object({
+      medium     = optional(string)
+      size_limit = optional(string)
+    }))
+    gcs = optional(object({
+      bucket    = string
+      read_only = optional(string)
+    }))
+    nfs = optional(object({
+      server    = string
+      path      = string
+      read_only = optional(string)
+    }))
+  }))
+  description = "[Beta] Volumes needed for environment variables (when using secret)"
+  default     = []
+}
+
+// Containers
+variable "containers" {
+  type = map(object({
```

Comment:
Can we make this as list(object) as we are doing for volumes input variable? @prabhu34

Reply:
Done